plugins/available/functional-test-plugin/index.js

```diff
@@ -42,9 +42,21 @@ class FunctionalTestPlugin {
     this.pipes['generic:document:afterDelete'] = (...args) => this.genericDocumentEvent('afterDelete', ...args);
   }
 
-  init (config, context) {
+  async init (config, context) {
     this.config = config;
     this.context = context;
+
+    // context.onDocument* related declarations ================================
+
+    const filters = { exists: 'name' };
+
+    await this.context.accessors.beforeDocumentWrite('test', 'test', filters, async (document, request) => {
```

Comment:
```suggestion
    await this.context.accessors.beforeDocumentWrite('test', 'test', filters, (document, request) => {
```

Reply:
This code is obsolete sorry, it was my first draft trying to implement the feature